cosmos/providers/dbt/core/operators/base.py

```diff
@@ -139,6 +126,8 @@ def __init__(
         dbt_ol_path = shutil.which("dbt-ol")
         if dbt_executable_path == "dbt" and shutil.which("dbt-ol"):
             self.dbt_executable_path = dbt_ol_path
+        elif dbt_executable_path == "dbt":
+            self.dbt_executable_path = shutil.which("dbt")
```

Comment:
How does this change relate to the profile refactor?
Perhaps we should introduce it in a different PR.

Reply:
This does relate to the env change - there was [incorrect code](https://github.com/astronomer/astronomer-cosmos/blob/1bce508f1f71553d33c72bee6aafb6169b3a4d0d/cosmos/providers/dbt/core/operators/base.py#L158-L160) on main that would append the system env if there was no `env` argument explicitly passed to the operator. When removing that, if you don't specify a dbt executable path, it would fail because it wasn't in the `PATH`.